hub/__init__.py

```diff
@@ -9,21 +9,28 @@
     "constants": False,
     "integrations": False,
     "tests": False,
+    "Dataset.clear_cache": False,
+    "Dataset.delete": False,
+    "Dataset.flush": False,
+    "Dataset.read_only": False,
+    "Dataset.size_approx": False,
+    "Dataset.token": False,
 }
 
 from .api.dataset import dataset
 from .api.read import read
 from .core.transform import compute, compose
+from .core.dataset import Dataset
 from .util.bugout_reporter import hub_reporter
 
 load = dataset.load
 empty = dataset.empty
 like = dataset.like
 list = dataset.list
-create_tensor = Dataset.create_tensor
-create_tensor_like = Dataset.create_tensor_like
-pytorch = Dataset.pytorch
-tensorflow = Dataset.tensorflow
+dataset_cl = Dataset
```

Comment:
not sure why we need dataset_cl to be exposed in the api

Reply:
To see the Dataset methods the docs.[
![Screenshot from 2021-08-05 14-35-49](https://user-images.githubusercontent.com/70504162/128336514-d728bf4c-dee7-419a-97bb-82dd92d2d17f.png)
](url)

Reply:
hmm, is there no other way to do that?

Reply:
I can move it to the core __init__.py, but then it will be shown on another page under hub.core.

Reply:
@istranic any thoughts here?

Reply:
Hmm, Dataset and dataset is a bit confusing, but it's better than putting it in hub.core. I think what you have above is fine. Two follow-ups:

Do we need to expose num_samples. Isn't that the same as len(ds)?

I've never used create_tensor_like. Pls make sure it's working and is not a bunch of TODOs. If it's incomplete, pls hide it for now.